src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -65,36 +70,51 @@ export async function chooseStorageAccount(diagnosticSettings: DiagnosticSetting
     const accountQuickPicks = storageAccountNameToStorageIdArray;
 
     // Create quick pick for more than 1 storage account scenario.
-    const selectedQuickPick = await vscode.window.showQuickPick(
-        accountQuickPicks,
-        {
-            placeHolder: "Select storage account for Periscope deployment:",
-            ignoreFocusOut: true
-        });
+    const selectedQuickPick = await vscode.window.showQuickPick(accountQuickPicks, {
+        placeHolder: "Select storage account for Periscope deployment:",
+        ignoreFocusOut: true,
+    });
 
     if (selectedQuickPick) {
         return selectedQuickPick.id;
     }
-
 }
 
 export async function getStorageInfo(
     kubectl: k8s.APIAvailable<k8s.KubectlV1>,
     cluster: AksClusterTreeItem,
     diagnosticStorageAccountId: string,
-    clusterKubeConfig: string
+    clusterKubeConfig: string,
 ): Promise<Errorable<PeriscopeStorage>> {
     try {
         const { resourceGroupName, name: accountName } = parseResource(diagnosticStorageAccountId);
 
         if (!resourceGroupName || !accountName) {
-            return { succeeded: false, error: `Invalid storage id ${diagnosticStorageAccountId} associated with the cluster` };
+            return {
+                succeeded: false,
+                error: `Invalid storage id ${diagnosticStorageAccountId} associated with the cluster`,
+            };
         }
 
         // Get keys from storage client.
-        const storageClient = new StorageManagementClient(cluster.subscription.credentials, cluster.subscription.subscriptionId);
+        const storageClient = new StorageManagementClient(
+            cluster.subscription.credentials,
+            cluster.subscription.subscriptionId,
+        );
         const storageAccKeyList = await storageClient.storageAccounts.listKeys(resourceGroupName, accountName);
-        const storageKey = storageAccKeyList.keys?.find((it) => it.keyName === "key1")?.value!;
+        if (storageAccKeyList.keys === undefined) {
```

Comment:
Apparently the previous code was "unsafe and wrong", which made me feel very ashamed of myself, so it's replaced by a bunch of very explicit checks.